satellite/audit/reporter.go

```diff
@@ -187,19 +189,29 @@ func (reporter *Reporter) recordPendingAudits(ctx context.Context, pendingAudits
 			}
 		} else {
 			// record failure -- max reverify count reached
-			_, err := reporter.overlay.UpdateStats(ctx, &overlay.UpdateRequest{
+			updateRequests = append(updateRequests, &overlay.UpdateRequest{
 				NodeID:       pendingAudit.NodeID,
 				IsUp:         true,
 				AuditSuccess: false,
 			})
-			if err != nil {
+		}
+	}
+
+	failedBatch, err := reporter.overlay.BatchUpdateStats(ctx, updateRequests)
+	if err != nil {
+		errlist.Add(err)
+	}
+	for _, nodeID := range failedBatch {
+		for _, pendingAudit := range pendingAudits {
+			if nodeID == pendingAudit.NodeID {
```

Comment:
maybe a better algorithm here would be to build a map of [NodeID]pendingAudit then check if any nodeIDs from failedBatch are present in the map, would decrease this in time complexity

Reply:
Added map for lookup